client.go

```diff
@@ -50,6 +51,14 @@ type PublishResult struct {
 	Offset    uint64
 }
 
+// EventHandler is a function to process the messages read from the stream and is passed as
+// a parameter to the subscribe call.
+type EventHandler = func(ctx context.Context, payload []byte, contentType string, headers map[string]string) error
```

Comment:
Consider changing payload type to Reader which *could* allow us to stream the payload in the future

Reply:
done!